src/Sulu/Component/Content/Document/Subscriber/StructureRemoveSubscriber.php

```diff
@@ -57,19 +58,19 @@ public function handleRemove(RemoveEvent $event)
 
     public function removeDocument($document)
     {
-        // TODO: This is not a good indicator. There should be a RoutableBehavior here.
-        if (!$document instanceof StructureBehavior) {
-            return;
-        }
-
         if ($document instanceof ChildrenBehavior) {
             foreach ($document->getChildren() as $child) {
                 $this->removeDocument($child);
             }
         }
 
-        $this->removeReferences($document);
-        $this->recursivelyRemoveRoutes($document);
+        if ($document instanceof StructureBehavior) {
+            $this->removeReferences($document);
+        }
+
+        if ($document instanceof RoutableBehavior) {
```

Comment:
@dantleech I know you've suggested that a few lines before, but is this really a good idea? Currently the very correct behavior would be to do that for every document being an instance of the `StructureBehavior` and having a property with the `sulu.rlp` tag, wouldn't it? Just introducing a marker interface here would feels somehow wrong... Because documents implementing the `RoutableBehavior` do not necessarily mean that they have routes (although it doesn't hurt if this method is called too often, excluding performance). Also that this Behavior is also used for this check feels wrong, because the name behavior implies that it is doing something, and used for more than a simple check.